File: src/main/java/com/implimentz/fragments/FragmentData.java
package com.implimentz.fragments;

/**
 * Created by Alexander Efremenkov.
 * Date: 07.01.16, 11:38
 * In Intellij IDEA 15.0.1 Ultimate
 * email: implimentz@gmail.com
 * twitter: iamironz
 */
final class FragmentData {

    private final String name;
    private final Fragment fragment;

    FragmentData(final String name, final Fragment fragment) {
        this.name = name;
        this.fragment = fragment;
    }

    final String getName() {
        return name;
    }
    final Fragment getFragment() {
        return fragment;
    }
}

File: src/main/java/com/implimentz/fragments/FragmentData.kt
package com.implimentz.fragments

import android.support.v7.app.AppCompatActivity

/**
 * Created by Alexander Efremenkov.
 * Date: 19.01.16, 14:57
 * In Intellij IDEA 15.0.1 Ultimate
 * email: implimentz@gmail.com
 * twitter: iamironz
 */
internal data class FragmentData<T, D: AppCompatActivity>(public val name: String, public val fragment: Fragment<T, D>)